legal-api/src/legal_api/services/filings/validations/common_validations.py

```diff
@@ -0,0 +1,71 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Common validations share through the different filings."""
+from legal_api.errors import Error
+
+
+def validate_share_structure(incorporation_json, filing_type) -> Error:  # pylint: disable=too-many-branches
+    """Validate the share structure data of the incorporation filing."""
+    share_classes = incorporation_json['filing'][filing_type] \
+        .get('shareStructure', {}).get('shareClasses', [])
+    msg = []
+    memoize_names = []
+
+    for index, item in enumerate(share_classes):
+        if item['name'] in memoize_names:
+            err_path = '/filing/{0}/shareClasses/{1}/name/'.format(filing_type, index)
+            msg.append({'error': 'Share class %s name already used in a share class or series.' % item['name'],
+                        'path': err_path})
+        else:
+            memoize_names.append(item['name'])
+
+        if item['hasMaximumShares']:
+            if not item.get('maxNumberOfShares', None):
+                err_path = '/filing/{0}/shareClasses/{1}/maxNumberOfShares/'.format(filing_type, index)
+                msg.append({'error': 'Share class %s must provide value for maximum number of shares' % item['name'],
+                            'path': err_path})
+        if item['hasParValue']:
+            if not item.get('parValue', None):
+                err_path = '/filing/{0}/shareClasses/{1}/parValue/'.format(filing_type, index)
+                msg.append({'error': 'Share class %s must specify par value' % item['name'], 'path': err_path})
+            if not item.get('currency', None):
+                err_path = '/filing/{0}/shareClasses/{1}/currency/'.format(filing_type, index)
+                msg.append({'error': 'Share class %s must specify currency' % item['name'], 'path': err_path})
+        for series_index, series in enumerate(item.get('series', [])):
```

Comment:
I might be tempted to break this series validation into a separate function , to make the linter happy, and also to break up this large function

Reply:
good suggestion, and easier to test too?

Reply:
Fixed.